README.md

```diff
@@ -104,14 +104,12 @@ spec:
       metadata: # optional, allows data source specific details to be passed
   timeWindow:
     # exactly one item, one of possible rolling time window or calendar aligned
-    # rolling time window
-    - unit: Second
-      count: numeric
+    ## rolling time window
+    - duration: duration-shorthand # duration of the window eg 1d, 28d
```

Comment:
```suggestion
    - duration: duration-shorthand # duration of the window, eg 1w, 28d; allowed time units: s, m, h, d, w
```

The argument around not allowing vague time units (like month, year) is correct and this this should be explicitly enshrined in the spec.

Reply:
Probably seconds are too small for the real-world scenario, the smallest unit should be minutes. What do you think?